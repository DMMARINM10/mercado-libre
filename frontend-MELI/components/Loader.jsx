const Loader = () => {
  return (
    <div className='loader-container'>
      <div data-testid={'loader'} className='loader'></div>
    </div>
  )
}

export default Loader
