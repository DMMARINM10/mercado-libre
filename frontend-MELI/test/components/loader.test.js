import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Loader from '@/components/Loader';

describe('Loader component', () => {
  it('data-testid loader is in the document', () => {
    render(<Loader />);
    const loader = screen.getByTestId('loader');
    expect(loader).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});