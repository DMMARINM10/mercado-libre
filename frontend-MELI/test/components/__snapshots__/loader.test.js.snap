// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader-container"
  >
    <div
      class="loader"
      data-testid="loader"
    />
  </div>
</DocumentFragment>
`;
